### ISSUE TITLE
Newsletter subscription success email not sent after confirmation

### ISSUE THREAD

**Issue by nfourteen** [magento/magento2#12439]
### Preconditions

1. Magento 2.2.1
2. Php 7.0

### Steps to reproduce

1. Change configuration under `Stores > Configuration > Customers > Newsletter > Subscription Options > Need to Confirm` to Yes
2. Sign up for Newsletter on frontend
3. Follow confirmation link in email

### Expected result

Should receive the Newsletter Subscription Success email.

### Actual result

No email is received, but database is changed from `STATUS_UNCONFIRMED` to `STATUS_SUBSCRIBED`.

### Proposed fix

I believe updating the following method will resolve the issue:

```
// Magento\Newsletter\Model\Subscriber::confirm($code) line 614

public function confirm($code)
{
    if ($this->getCode() == $code) {
        $this->setStatus(self::STATUS_SUBSCRIBED)
            ->setStatusChanged(true)
            ->save();
        return true;
    }
    return false;
}
```

Should be:

```
public function confirm($code)
{
    if ($this->getCode() == $code) {
        $this->setStatus(self::STATUS_SUBSCRIBED)
            ->setStatusChanged(true)
            ->save();

        $this-> sendConfirmationSuccessEmail();
        return true;
    }
    return false;
}
```

**Comment by magento-engcom-team** [comment]
@nfourteen, thank you for your report.
We've created internal ticket(s) MAGETWO-84525 to track progress on the issue.

**Comment by RhodriOwainDavies** [comment]
Hi I've started work on this issue, can you assign to me please?

**Comment by Styopchik** [comment]
#mageconf

**Comment by magento-engcom-team** [comment]
@Styopchik thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by okorshenko** [comment]
Hi @nfourteen. Thank you for your report. 
The issue has been fixed in magento/magento2#12751 by @Styopchik in 2.2-develop branch
Related commit(s):
 - c4e4168cdeb349bd45fcb609970ab1219231c8f7

The fix will be available with the upcoming patch release.
